tests/functional/adapter/test_custom_schema.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+from dbt.tests.util import run_dbt, run_sql_with_adapter
+
+seed_csv = """
+id,name,date
+1,Easton,1981-05-20 06:46:51
+2,Lillian,1978-09-03 18:10:33
+3,Jeremiah,1982-03-11 03:59:51
+4,Nolan,1976-05-06 20:21:35
+""".lstrip()
+
+
+class TestCustomSchemaBase:
+    """
+    This test is meant to ensure that Trino table, view, incremental materialization
+    works as expected for custom schemas
+    """
+
+    # Setting __test__ to False, to avoid testing base class
+    __test__ = False
+
+    # set custom schema name
+    custom_schema_name = "very_custom_schema_name"
+
+    # define model
+    def custom_schema_model(self, materialization, custom_schema_name):
+        return f"""
+                    {{{{
+                        config(
+                        materialized = "{materialization}",
+                        schema="{custom_schema_name}"
+                        )
+                    }}}}
+                    select * from {{{{ ref('seed') }}}}
+                """
+
+    @property
+    def schema(self):
+        return "default"
+
+    # everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "seed.csv": seed_csv,
+        }
+
+    # everything that goes in the "models" directory
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            f"custom_schema_{self.materialization}_model.sql": self.custom_schema_model(
+                self.materialization, self.custom_schema_name
+            )
+        }
+
+    def test_custom_schema_trino(self, project):
+        # Seed seeds, run models.
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 1
+        results = run_dbt(["run"], expect_pass=True)
+        assert len(results) == 1
+
+        # Fetch info from information_schema about just created table/view.
+        sql = f"""
+            select * from {project.adapter.config.credentials.database}.information_schema.tables
+            where table_catalog = '{project.adapter.config.credentials.database}'
+            and table_schema = '{project.adapter.config.credentials.schema}_{self.custom_schema_name}'
+        """.strip()
+        results = run_sql_with_adapter(project.adapter, sql, fetch="all")
+
+        # Check if fetched info is as expected to be.
+        assert results[0][0] == project.adapter.config.credentials.database
+        assert (
+            results[0][1]
+            == f"{project.adapter.config.credentials.schema}_{self.custom_schema_name}"
+        )
+        assert results[0][2] == f"custom_schema_{self.materialization}_model"
+        assert results[0][3] == self.table_type
```

Comment:
I would make this an abstract property that inherited classes need to implement (check abc)